docs/extend/extend-apiml/api-mediation-oidc-authentication.md

```diff
@@ -0,0 +1,86 @@
+The following diagram illustrates the interactions between the participants of the OIDC-based API ML authentication process.
+
+<img src={require("../../images/api-mediation/apiml-oidc-auth-seq.png").default} alt="APIML OIDC Workflow" width="700"/>
+
+* When the User wants to access mainframe resources or services using the client application without valid authentication or an access token, the client redirects the User agent to the login end-point of the distributed OIDC provider. 
+* The user is asked to provide valid credentials (authentication factors).
+* After successful validation of all authentication factors, the OIDC provider grants an access token to the client.
+* The client can then request the needed mainframe resources and present the access token from the API ML Gateway in the request. 
+* The Gateway validates the access token at the OIDC provider's /introspection end-point. If successful, the outcome is cached for a short time. 
+* In subsequent calls with the same token, the Gateway reuses the cached validation outcome. This outcome reuse reduces round trips to the OIDC /introspection end-point for a short interval of time when the client needs to access multiple resources to complete a unit of work. The caching interval is configurable with a sensible default. 
+* The API ML Gateway fetches the distributed user identity from the distributed access token, and maps that identity to the mainframe identity of the user using SAF. 
+* The mainframe user identity is used to create the mainframe user credentials (Zowe or SAF JWT or pass-ticket) expected by the target mainframe service.
+
+## API ML OIDC Configuration
+
+### Prerequisites
+
+- Users who need access to mainframe resources using OIDC authentication require a mainframe identity managed by SAF/ESM
+- Users require a distributed identity managed by the OIDC provider  
+- SAF/ESM is configured with mapping between the mainframe and distributed user identities
+  
+### OIDC provider configuration
+
+- Client Application configuration in the OIDC provider. 
+
+  Configuration of the OIDC provider varies depending on the OIDC provider and client application capabilities.
+  For example, web applications with a secure server-side component can use the `code grant authorization flow`, and can be granted a Refresh Token. Conversely, a single page application running entirely in the user agent (browser) is more limited regarding its security capabilities.  
+
+  Consult your OIDC provider documentation for options and requirements available for your type of client application. 
+
+- Users assignment to the Client Application.
+
+  Users who are allowed access the mainframe resources with a distributed authentication must either be directly assigned by the OIDC provider to the client application,
+or must be part of a group which is allowed to work with the client application.     
+
+### API ML Gateway configuration
+
+* Allow OIDC authentication in zowe.yaml
+  * set oidc_allowed
```

Comment:
Does this really work?

Reply:
Fixed with property components.gateway.apiml.security.oidc.enabled, and other similar.